app/models/notification.rb

```diff
@@ -0,0 +1,78 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+
+  has_one :activity, inverse_of: :notification, dependent: :destroy
+
+  # Send notifications according to input types and objects
+  #
+  # @param [Object] recipient who does these notifications send for
+  # @param [Object] sender who generate this notification
+  # @param [Object] object what is this notification about
+  # @param [Hash] types which kinds of notifications that will be sent
+  # @option types [Symbol] :activity the type name of activity
+  # @option types [Boolean] :popup enable to send popup notification
+  # @option types [Boolean] :email enable to send email notification
+  def self.notify(recipient, sender, object, types =
+                  { activity: false, popup: false, email: false })
```

Comment:
hmm... first it is hard to describe what can be in types array in yard doc. Second... activity should take in a symbol not a boolean. So that will be `types: [{activity: :created}, :popup ]` weird right?